pkg/xds/context/aggregate_mesh_context.go

```diff
@@ -32,15 +32,28 @@ func AggregateMeshContexts(
 		meshContextsByName[mesh.Meta.GetName()] = meshCtx
 	}
 
+	hash := aggregatedHash(meshContexts)
+
 	egressByName := map[string]*core_mesh.ZoneEgressResource{}
 	if len(meshContexts) > 0 {
 		for _, egress := range meshContexts[0].Resources.ZoneEgresses().Items {
 			egressByName[egress.Meta.GetName()] = egress
 		}
+	} else {
+		var egressList core_mesh.ZoneEgressResourceList
```

Comment:
Why are we only doing this for egress? I thought the same problems were with ingress.

Reply:
Only ZoneEgresses are listed in AggregatedMeshContexts and ZoneIngress builder needs ZoneEgresses to properly build itc config